extension/app/src/pages/MainPage.tsx

```diff
@@ -1,40 +1,126 @@
 import {
   Button,
   ExternalLinkIcon,
+  LoginIcon,
   LogoHorizontalColorLogo,
-  MarkPenIcon,
+  LogoutIcon,
   Page,
-  TranslateIcon,
+  Spinner,
+  TextArea,
 } from "@dust-tt/sparkle";
-import { Link } from "react-router-dom";
+import { useCallback, useEffect, useState } from "react";
+
+import { sendAuthMessage, sentLogoutMessage } from "../lib/auth";
+import {
+  clearAccessToken,
+  getAccessToken,
+  saveAccessToken,
+} from "../lib/utils";
 
 export const MainPage = () => {
+  const [token, setToken] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
```

Comment:
`nit`
As this will be needed in many different places to wrap the calls, I'd suggest creating a custom hook `useAuth` that bundles all this logic.

Reply:
💯 done!